it/src/main/scala/com/wavesplatform/it/MatcherNode.scala

```diff
@@ -0,0 +1,60 @@
+package com.wavesplatform.it.transactions
+
+import com.wavesplatform.account.PrivateKeyAccount
+import com.wavesplatform.it.Nodes
+import com.wavesplatform.it.util._
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.transaction.smart.SetScriptTransaction
+import com.wavesplatform.transaction.smart.script.ScriptCompiler
+import com.wavesplatform.utils.ScorexLogging
+import org.scalatest.{BeforeAndAfterAll, Suite}
+import play.api.libs.json.JsNumber
+
+trait MatcherNode extends BeforeAndAfterAll with Nodes with ScorexLogging {
+  this: Suite =>
+
+  def matcherNode = nodes.head
+  def aliceNode   = nodes(1)
+  def bobNode     = nodes(2)
+
+  protected lazy val matcherAddress: String = matcherNode.createAddress()
+  protected lazy val aliceAddress: String   = aliceNode.createAddress()
+  protected lazy val bobAddress: String     = bobNode.createAddress()
+
+  protected lazy val matcherAcc = PrivateKeyAccount.fromSeed(matcherNode.seed(matcherAddress)).right.get
+  protected lazy val aliceAcc   = PrivateKeyAccount.fromSeed(aliceNode.seed(aliceAddress)).right.get
+  protected lazy val bobAcc     = PrivateKeyAccount.fromSeed(bobNode.seed(bobAddress)).right.get
+
+  private val addresses = Seq(matcherAddress, aliceAddress, bobAddress)
+
+  //really before all tests, because FreeSpec issue with "-" and "in"
+  initialBalances()
```

Comment:
Actually it's a bad idea, because we should think about traits linearisation in tests. E.g.:
```scala
trait Foo { println("foo") }
trait Bar { println("bar") }
trait Baz {}

class X extends Baz with Foo with Bar {}
class Y extends Baz with Bar with Foo {}

new X
// foo
// bar

new Y
// bar
// foo
```

So, if someone write a similar class and mix it with `MatcherNode`, it will face bizarre problems.